### PR TITLE
RAD-218 Add tests for RadiologyReportValidator

### PR DESCRIPTION
**Pull request by joehiggs**
Make sure the supports(Class c) returns
* true for objects of the RadiologyReport class
* false for other types

Make sure the validate(Object obj, Errors errors)
* fails when
** radiology report is null
** principalResultsInterpreter is empty or whitespace
* passes when
** all fields are correct

See https://issues.openmrs.org/browse/RAD-218

### CODE REVIEW

File: api/src/test/java/org/openmrs/module/radiology/validator/RadiologyReportValidatorComponentTest.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	
+	/**
+	 * @see {@link RadiologyReportValidator#supports(Class c)}
+	 */
+	@Test
+	@Verifies(value = "should return true for RadiologyReport objects", method = "supports(Class c)")
+	public void supports_shouldReturnTrueForRadiologyReportObjects() throws Exception {
+		
+		RadiologyReportValidator radiologyReportValidator = new RadiologyReportValidator();
+		Assert.assertTrue(radiologyReportValidator.supports(RadiologyReport.class));
```
Comment (teleivo) [inline api/src/test/java/org/openmrs/module/radiology/validator/RadiologyReportValidatorComponentTest.java]: please use static imports for all asserts

**Comment by teleivo** [comment]
@joehiggs please use the https://wiki.openmrs.org/display/docs/Generate+Test+Case+Plugin
this will create the unit tests in the correct style just using your "@should xxx" description you write on the methods to be tested. You wrote your unit tests in the former style. I know this is confusing.

File: api/src/test/java/org/openmrs/module/radiology/validator/RadiologyReportValidatorComponentTest.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+		Study study = new Study();
+		study.setPerformedStatus(PerformedProcedureStepStatus.COMPLETED);
+		radiologyOrder.setStudy(study);
+		RadiologyReport radiologyReport = new RadiologyReport(radiologyOrder);
+		
+		Errors errors = new BindException(radiologyReport, "radiologyReport");
+		new RadiologyReportValidator().validate(null, errors);
+		
+		Assert.assertTrue(errors.hasErrors());
+		Assert.assertEquals("error.general", ((List<ObjectError>) errors.getAllErrors()).get(0)
```
Comment (teleivo) [inline api/src/test/java/org/openmrs/module/radiology/validator/RadiologyReportValidatorComponentTest.java]: please use the hamcrest style assertions for anything beside true/false checks. like for ex. here https://github.com/openmrs/openmrs-module-radiology/blob/1419ac34d12b4a869eb6a309f39e71ed2e6615b2/api/src/test/java/org/openmrs/module/radiology/dicom/DicomWebViewerTest.java#L45

**Comment by teleivo** [comment]
@joehiggs you have forked my fork repo. I think its better if you close your PRs. Fork the real upstream repo https://github.com/openmrs/openmrs-module-radiology and open them again.

**Comment by teleivo** [comment]
RadiologyReportValidatorComponentTest is not a component test, its a unit test since it does not extend BaseModuleContextSensitiveTest. You are not testing the code + DB. Please just rename RadiologyReportValidatorComponentTest --> RadiologyReportValidatorTest

**Comment by teleivo** [comment]
commit title should also be formatted this way "RAD-218 Add tests for RadiologyReportValidator"

**Comment by teleivo** [comment]
@joehiggs thanks a lot for your effort! please update the PR with my comments and I'll merge it. and please dont despair with my nitpicking, but being consistent adds a lot to our code quality :+1:

File: api/src/test/java/org/openmrs/module/radiology/validator/RadiologyReportValidatorComponentTest.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+import org.openmrs.module.radiology.RadiologyOrder;
+import org.openmrs.module.radiology.Study;
+import org.openmrs.module.radiology.report.RadiologyReport;
+import org.openmrs.test.Verifies;
+import org.springframework.validation.BindException;
+import org.springframework.validation.Errors;
+import org.springframework.validation.ObjectError;
+
+/**
+ * Tests methods on the {@link RadiologyReportValidator} class.
```
Comment (teleivo) [inline api/src/test/java/org/openmrs/module/radiology/validator/RadiologyReportValidatorComponentTest.java]: just write "Tests {xxx}."

**Comment by joehiggs** [comment]
@teleivo Thanks for the input! I've made all the corrections to the files; however, I'm not entirely sure how I should go about changing my fork without losing my branches. Do you have any advice for how I should go about that?

**Pull request closed without merging**